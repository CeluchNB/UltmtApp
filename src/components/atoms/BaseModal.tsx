import React from 'react'
import { useColors } from '../../hooks'
import { Modal, StyleSheet, View } from 'react-native'

interface BaseModalProps {
    visible: boolean
    onClose: (...args: any) => void
}

const BaseModal: React.FC<BaseModalProps> = props => {
    const { visible, children, onClose } = props
    const { colors } = useColors()

    const styles = StyleSheet.create({
        modalContainer: {
            flex: 1,
            justifyContent: 'center',
            alignItems: 'center',
            display: 'flex',
            flexDirection: 'row',
            backgroundColor: `${colors.primary}50`,
        },
        modalView: {
            backgroundColor: colors.primary,
            borderColor: colors.textPrimary,
            color: colors.textPrimary,
            borderRadius: 10,
            padding: 25,
            alignItems: 'center',
            shadowColor: colors.textPrimary,
            elevation: 5,
            flex: 1,
            margin: 20,
        },
    })
    return (
        <Modal
            animationType="slide"
            visible={visible}
            onRequestClose={onClose}
            transparent={true}
            testID="base-modal">
            <View style={styles.modalContainer}>
                <View style={styles.modalView}>{children}</View>
            </View>
        </Modal>
    )
}

export default BaseModal
