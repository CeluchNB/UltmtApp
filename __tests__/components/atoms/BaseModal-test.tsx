import BaseModal from '../../../src/components/atoms/BaseModal'
import React from 'react'
import { View } from 'react-native'
import { render } from '@testing-library/react-native'

it('should match snapshot with minimal children', () => {
    const snapshot = render(
        <BaseModal visible={true} onClose={jest.fn()}>
            <View />
        </BaseModal>,
    )

    expect(snapshot).toMatchSnapshot()
})
