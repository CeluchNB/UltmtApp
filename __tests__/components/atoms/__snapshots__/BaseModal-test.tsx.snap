// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with minimal children 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  testID="base-modal"
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#30303050",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1a1a1a",
          "borderColor": "#ffd600",
          "borderRadius": 10,
          "color": "#ffd600",
          "elevation": 5,
          "flex": 1,
          "margin": 20,
          "padding": 25,
          "shadowColor": "#ffd600",
        }
      }
    >
      <View />
    </View>
  </View>
</Modal>
`;
